Library/Homebrew/requirements/perl_requirement.rb

```diff
@@ -0,0 +1,28 @@
+class PerlRequirement < Requirement
+  fatal true
+  default_formula "perl"
+
+  def initialize(tags)
+    @version = tags.shift if /(\d.\d{1,2})/ === tags.first
+    raise "PerlRequirement requires a version!" unless @version
+    super
+  end
+
+  satisfy(:build_env => false) do
+    which_all("perl").detect do |perl|
+      perl_version = /(\d\.\d(\d)?)/.match Utils.popen_read(perl, "--version")
```

Comment:
Looking at the `perl --version` output I think it's worth making the match a bit stricter, too (a bit like in the other comment). I think this should have essentially the same effect, but be more robust:

```ruby
perl_version = Utils.popen_read(perl, "--version")[/\(v(\d+\.\d+)(?:\.\d+)?\)/, 1]
```

This will match the version inside `(v5.16.3)` when seeing a version line like `This is perl 5, version 16, subversion 3 (v5.16.3) built for darwin-thread-multi-2level` and return `5.16`.

It also works if the output should happen to be just `(v5.16)`.